File: examples/fsm_platformer/fsm_platformer_unity/LandingTimerTransition.cs
// Unity APIs used: none (reads FSMState.TimeState)
public class LandingTimerTransition : FSMTransition
{
    private const float LandingDuration = 0.1f;

    public LandingTimerTransition(FSMState targetState) : base(targetState) { }

    public override bool IsValid(FSMState currentState)
    {
        return currentState.TimeState >= LandingDuration;
    }
}

File: examples/fsm_platformer/fsm_platformer_python/landing_timer_transition.py
"""Landing Timer Transition — LANDING -> IDLE after 0.1s."""

from __future__ import annotations

from fsm_platformer_python.fsm import FSMState, FSMTransition

LANDING_DURATION = 0.1


class LandingTimerTransition(FSMTransition):
    """Fires when the landing state has lasted at least LANDING_DURATION seconds."""

    def __init__(self, target_state: FSMState) -> None:
        super().__init__(target_state)

    def is_valid(self, current_state: FSMState) -> bool:
        return current_state.time_state >= LANDING_DURATION
